src/validators/StringValidator.ts

```diff
@@ -29,6 +37,28 @@ export class StringValidator<T extends string> extends BaseValidator<T> {
 		return this.addConstraint(stringLengthNe(length) as IConstraint<T>);
 	}
 
+	public email(): this {
+		const emailRegex = /^(([^<>()[\]\.,;:\s@\"]+(\.[^<>()[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i;
```

Comment:
Do you even know what this RegExp does? If so, are you aware that the URL validation in here doesn't match the RegExp you used in the url method?

It should at least have a link to the source where you got the RegExp from, and it should also be moved outside the method, there isn't a need to create a new RegExp instance over and over.

Reply:
> Do you even know what this RegExp does? If so, are you aware that the URL validation in here doesn't match the RegExp you used in the url method?

I'd added it at last that's why I didn't changed the url regex. Ik this checks for all special characters in domain name but in url I'd only added check for space. I'll remove the url regex so no need to change that now.

Reply:
changed